test/integration/oidc/oidc_test.go

```diff
@@ -0,0 +1,435 @@
+	apiServer := startTestAPIServerForOIDC(t, oidcServer.URL(), defaultOIDCClientID, caFilePath)
+	t.Cleanup(func() {
+		apiServer.TearDownFn()
+		oidcServer.TearDownFn()
+	})
+
+	adminClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+	configureRBAC(t, adminClient, defaultRole, defaultRoleBinding)
+
+	oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(oidcTest.DefaultTokenHandlerBehaviour(
+		t,
+		rsaPrivateKeyFilePath,
+		oidcServer.URL(),
+		defaultOIDCClientID,
+		defaultOIDCClaimedUsername,
+		defaultStubAccessToken,
+		defaultStubRefreshToken,
+		time.Now().Add(time.Second*1200).Unix(),
+	))
+
+	idToken, stubRefreshToken := fetchOIDCCredentials(t, oidcServer.TokenURL(), caFilePath)
+	configureClientConfigForOIDC(apiServer.ClientConfig, idToken, stubRefreshToken, oidcServer.URL())
+
+	client := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+
+	oidcServer.JwksHandler().EXPECT().KeySet().AnyTimes().DoAndReturn(oidcTest.DefaultJwksHandlerBehaviour(t, "./testdata/rsa2.pub"))
+
+	_, err := client.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+	targetErr := new(errors.StatusError)
+	assert.ErrorAs(t, err, &targetErr)
+}
+
+func configureTestInfrastructure(t *testing.T) (*oidcTest.TestServer, *kubeAPIServerTesting.TestServer) {
+	t.Helper()
+
+	oidcServer := oidcTest.BuildAndRunTestServer(t, caFilePath, caKeyFilePath)
+	apiServer := startTestAPIServerForOIDC(t, oidcServer.URL(), defaultOIDCClientID, caFilePath)
+	t.Cleanup(func() {
+		apiServer.TearDownFn()
+		oidcServer.TearDownFn()
+	})
+
+	oidcServer.JwksHandler().EXPECT().KeySet().AnyTimes().DoAndReturn(oidcTest.DefaultJwksHandlerBehaviour(t, rsaPublicKeyFilePath))
+
+	adminClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+	configureRBAC(t, adminClient, defaultRole, defaultRoleBinding)
+
+	return oidcServer, apiServer
+}
+
+func configureRBAC(t *testing.T, clientset *kubernetes.Clientset, role *rbacV1.Role, binding *rbacV1.RoleBinding) {
+	t.Helper()
+
+	_, err := clientset.RbacV1().Roles(defaultNamespace).Create(context.TODO(), role, metaV1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = clientset.RbacV1().RoleBindings(defaultNamespace).Create(context.TODO(), binding, metaV1.CreateOptions{})
+	require.NoError(t, err)
+}
+
+func configureClientConfigForOIDC(config *restClient.Config, idToken, refreshToken, oidcServerURL string) {
+	config.BearerToken = ""
+	config.AuthProvider = &api.AuthProviderConfig{
+		Name: "oidc",
+		Config: map[string]string{
+			"client-id":                 defaultOIDCClientID,
+			"id-token":                  idToken,
+			"idp-issuer-url":            oidcServerURL,
+			"idp-certificate-authority": caFilePath,
+			"refresh-token":             refreshToken,
+		},
+	}
+}
+
+func startTestAPIServerForOIDC(t *testing.T, oidcURL, oidcClientID, oidcCAFilePath string) *kubeAPIServerTesting.TestServer {
+	t.Helper()
+
+	return kubeAPIServerTesting.StartTestServerOrDie(
```

Comment:
use `k8s.io/kubernetes/test/integration/framework.StartTestServer()`

Reply:
this was not addressed yet

Reply:
It was done. I have no idea why you saw not updated version, probably due to I forgot submit PR review. Please review again

Reply:
I still see the old code

Reply:
No idea, but as far as I see this comment is labeled as outdated.
![Screenshot from 2023-02-25 18-58-30](https://user-images.githubusercontent.com/6449847/221366843-c91aefce-8454-4b97-a03a-db39c455a4d5.png)

Maybe you should check again. Currently `kubeapiserverapptesting.StartTestServer` is being used